src/test/java/seedu/address/logic/commands/contacts/DeleteContactByTagCommandTest.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.commands.contacts;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Calendar;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.RemindersImpl;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.TagTreeImpl;
+
+/**
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for
+ * {@code DeleteContactCommand}.
+ */
+public class DeleteContactByTagCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new Calendar(),
+            new TagTreeImpl(), new UserPrefs(), new RemindersImpl());
+
+    @Test
+    public void execute_deleteContactByTag_success() {
+        Tag tagForDeletion = new Tag("friends");
+        DeleteContactByTagCommand deleteContactByTagCommand = new DeleteContactByTagCommand(tagForDeletion);
+
+        String expectedMessage = DeleteContactByTagCommand.MESSAGE_DELETE_PERSON_SUCCESS + tagForDeletion.tagName;
+
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(),
+                new Calendar(), new TagTreeImpl(), new UserPrefs(), new RemindersImpl());
```

Comment:
Consider using ModelManagerBuilder class instead of invoking the constructor directly.